src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -382,12 +384,14 @@ public UserDetails call() throws AuthenticationException, NamingException {
                         // locate this user's record
                         final String domainDN = toDC(domain.getName());
 
-                        Attributes user = new LDAPSearchBuilder(context, domainDN).subTreeScope().searchOne("(& (userPrincipalName={0})(objectCategory=user))", userPrincipalName);
+                        Attributes user = new LDAPSearchBuilder(context, domainDN).subTreeScope().returns("*", "+")
+                                .searchOne("(& (userPrincipalName={0})(objectCategory=user))", userPrincipalName);
```

Comment:
I'm no AD admin, but theoretically, these two LDAP filters can be combined into one like so:

`(&(objectCategory=user)(|(userPrincipalName={0})(sAMAccountName={0})))`

No idea how that would affect or potentially destroy performance of LDAP indices that may be set up.